Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -239,43 +216,47 @@ public class MpcCoreKit {
             rss_comm: rss_comm)
 
         let key_details = try await tkey!.initialize(never_initialize_new_key: false, include_local_metadata_transitions: false)
-        appState.metadataPubKey = try key_details.pub_key.getPublicKey(format: .EllipticCompress)
-        
+        metadataPubKey = try key_details.pub_key.getPublicKey(format: .EllipticCompress)
+
         if key_details.required_shares > 0 {
```

Comment:
<img width="659" alt="Screenshot 2024-12-03 at 9 59 43 AM" src="https://github.com/user-attachments/assets/8121f84d-f5cd-4761-9341-6f0dd8d72b00">

We should check if user is new or old based on if metadata for oauthKey is available or not as done in web.

Checking based on required_shares does not seems correct

Reply:
@ieow

Reply:
checking the require share can determine existing user or not.
as for existing user tkey always initalized with single share only. It always need extra share from the device factor or hash factor

for new user, the require share is always 0 because new created tkey always have 2 shares